Library/Homebrew/rubocops/cask/ast/stanza.rb

```diff
@@ -52,6 +51,19 @@ def same_group?(other)
           stanza_group == other.stanza_group
         end
 
+        def comments
+          @comments ||= stanza_node.each_node.reduce([]) do |comments, node|
+            comments | comments_hash[node.loc]
+          end
+        end
+
+        def comments_hash
+          @comments_hash ||= Parser::Source::Comment.associate_locations(
+            stanza_node.parent,
+            comments,
```

Comment:
This `comments` was previously `CaskBlock#comments`, not `Stanza#comments`, which is why it's an infinite loop now.

Reply:
So we _can't_ move these methods in here since `Stanza` doesn't know about all of `CaskBlock` and its `comments` it needs to know about?

Reply:
It's OK, I figured it out (I hope). 🧠💡

Reply:
But is `CaskBlock#comments` correct for nested comments? Shouldn't it be something like `stanza_node.parent.comments`?

Reply:
The tests that were failing before now pass... 🤷🏻 Let me have a look some more!

Reply:
Ohhh right, I see what you mean, *nested* comments in `on_*` blocks - we don't have any tests that cover those.

Reply:
🤔 But if we have `stanza_node.parent.comments` for nested comments (that `.comments` method like that doesn't work btw) how do we handle the normal, non-nested, comments? Maybe I'm too tired for this right now. 😓

Reply:
For the non-nested ones `stanza_node.parent.comments` (or equivalent method that works) should just be the same as before, since the cask block is their parent, no?

Reply:
That makes sense. I clearly was too tired. 😂

Reply:
Turns out I was trying to do too much, the `StanzaOrder` test failing with the infinite loop was the thing that needed fixing, not the `StanzaOrder` handling of nested comments (since that's a separate cop, huh). I backed one of the failing tests out, concentrated on `StanzaGrouping` and will fix the (pre-existing) TODOs about nested `StanzaOrder` detection in a separate PR.